openff/evaluator/storage/localfile.py

```diff
@@ -44,7 +58,29 @@ def _store_object(
 
             shutil.move(ancillary_data_path, directory_path)
 
+        if self._cache_objects_in_memory:
+            self._cached_retrieved_objects[storage_key] = (
+                object_to_store,
+                ancillary_data_path,
+            )
+
     def _retrieve_object(self, storage_key, expected_type=None):
+        if (
+            self._cache_objects_in_memory
+            and storage_key in self._cached_retrieved_objects
+        ):
+            return self._cached_retrieved_objects[storage_key]
+        stored_object, ancillary_path = self._retrieve_object_from_storage(
+            storage_key, expected_type
+        )
```

Comment:
Can `storage_key` meaningfully be `None`? The type annotation (already existing, not in the changeset) is throwing me off

Reply:
I don't think so, it's consistently annotated as a string. If it's provided as None in other methods, a new str key is generated. The `expected_type` is provided, that's the expected type of the object stored with the `storage_key`.